lib/honeybadger/plugins/active_job.rb

```diff
@@ -37,7 +37,9 @@ def context(job) # rubocop:disable Metrics/MethodLength
         requirement do
           defined?(::Rails.application) &&
             ::Rails.application.config.respond_to?(:active_job) &&
-            !EXCLUDED_ADAPTERS.include?(::Rails.application.config.active_job[:queue_adapter])
+            (queue_adapter = ::Rails.application.config.active_job[:queue_adapter]) &&
+            !EXCLUDED_ADAPTERS.include?(queue_adapter.to_sym) &&
+            !(defined?(::GoodJob) && ::GoodJob.on_thread_error.nil? && queue_adapter.to_sym == :good_job) # Don't report errors if GoodJob is reporting them
```

Comment:
@stympy shouldn't the `on_thread_error` condition be flipped? 🤔 
```ruby
!(defined?(::GoodJob) && !::GoodJob.on_thread_error.nil? && queue_adapter.to_sym == :good_job)
```

Reply:
Good catch! Yes, it should, though the way that it's written works due to ... reasons. :) The fix is [here](https://github.com/honeybadger-io/honeybadger-ruby/pull/554).